Fixed issue where SSSD wasn't started after an auth configmap update.

abellotti opened the pull request:
- Attempt to enable and starting during initialize-httpd-auth service is too early in the process.
- Created an enable-sssd service which we run after both initialize-httpd-auth and httpd which enables and start sssd if the /etc/sssd/sssd.conf file was overlayed.
- Simplified the logic in initialize-httpd-auth since that was moved out.

abellotti committed:
Fixed issue where SSSD wasn't started after an auth configmap update.

- Attempt to enable and starting during initialize-httpd-auth service
  is too early in the process.
- Created an enable-sssd service which we run after both initialize-httpd-auth
  and httpd which enables and start sssd if the /etc/sssd/sssd.conf file was
  overlayed.
- Simplified the logic in initialize-httpd-auth since that was moved out.

abellotti commented:
Needed this fix for the ldap (ext-auth w/sssd) tests in https://github.com/ManageIQ/manageiq-pods/pull/210

carbonin commented:
Can we use a conditional in the sssd unit on sssd.conf instead of trying to delay the start up?

Or is it okay to just always enable it and let it fail if the file isn't present?

abellotti commented:
I think so, I can add a ConditionPathExists on the sssd.service Unit and always enable it. As long as systemd does not do the unit condition checks until it is time to start the service, it should work.  I'll give that a try, Thanks !!.

abellotti committed:
Updated sssd startup issues by always enabling the service but
adding conditions based on the initialize-httpd-auth service and
the /etc/sssd/sssd.conf file presence.

abellotti commented:
@carbonin all good, nothing a little After and ConditionPathExists couldn't help.

![screen shot 2017-09-08 at 11 27 41 am](https://user-images.githubusercontent.com/5797328/30220737-802748ea-948e-11e7-8138-b52a615d891d.png)

![screen shot 2017-09-08 at 11 54 25 am](https://user-images.githubusercontent.com/5797328/30220740-8559e30e-948e-11e7-9365-86cacf6528d0.png)

diffs: https://github.com/ManageIQ/container-httpd/pull/16/commits/8ffa545b4109cd3601a2f0612799d5bbf81ab7b6

carbonin commented on Dockerfile:
```diff
@@ -69,6 +69,10 @@ COPY docker-assets/initialize-httpd-auth.sh   /usr/bin
 
+## Make sure sssd has the right startup conditions
+RUN  mkdir -p /etc/systemd/system/sssd.service.d
+COPY docker-assets/sssd-startup.conf      /etc/systemd/system/sssd.service.d/startup.conf
```
Why the extra spacing here?

abellotti replied:
sorry, was aligned with the lower COPY, will update.

carbonin reviewed (approved):
Looks good to me other than the one extra spacing.

abellotti committed:
Fixed spacing in Dockerfile
